src/DataProvider/UserBookingCollectionDataProvider.php

```diff
@@ -5,16 +5,18 @@
 use ApiPlatform\Core\DataProvider\ContextAwareCollectionDataProviderInterface;
 use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
 use App\Entity\Main\UserBooking;
+use App\Security\Voter\UserBookingVoter;
 use App\Service\MicrosoftGraphServiceInterface;
 use Exception;
 use GuzzleHttp\Exception\GuzzleException;
 use Microsoft\Graph\Exception\GraphException;
 use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\Security\Core\Security;
 use Symfony\Component\Uid\Ulid;
 
 final class UserBookingCollectionDataProvider implements ContextAwareCollectionDataProviderInterface, RestrictedDataProviderInterface
 {
-    public function __construct(private MicrosoftGraphServiceInterface $microsoftGraphService)
+    public function __construct(private MicrosoftGraphServiceInterface $microsoftGraphService, private Security $security)
```

Comment:
All constructor parameters should be marked readonly (See https://php.watch/versions/8.1/readonly#constructor-readonly)